drivers/watchdog/wdt_opentitan.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023 by Rivos Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT lowrisc_opentitan_aontimer
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/watchdog.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_opentitan, CONFIG_WDT_LOG_LEVEL);
+
+struct ot_aontimer_regs {
+	/* 0x00 */
+	uint32_t alert_test;
+	/* 0x04 */
+	uint32_t wkup_ctrl;
+	/* 0x08 */
+	uint32_t wkup_thold;
+	/* 0x0C */
+	uint32_t wkup_count;
+	/* 0x10 */
+	uint32_t wdog_regwen;
+	/* 0x14 */
+	uint32_t wdog_ctrl;
+	/* 0x18 */
+	uint32_t wdog_bark_thold;
+	/* 0x1C */
+	uint32_t wdog_bite_thold;
+	/* 0x20 */
+	uint32_t wdog_count;
+	/* 0x24 */
+	uint32_t intr_state;
+} __packed;
+
+struct wdt_ot_aontimer_cfg {
+	volatile struct ot_aontimer_regs *regs;
+	uint32_t clk_freq;
+	bool wdog_lock;
+};
+
+struct wdt_ot_aontimer_data {
+	wdt_callback_t bark;
+	bool en;
+};
+
+static int ot_aontimer_setup(const struct device *dev, uint8_t options)
+{
+	const struct wdt_ot_aontimer_cfg *const cfg = dev->config;
+	volatile struct ot_aontimer_regs *regs = cfg->regs;
+
+	sys_write32(0, (mem_addr_t) &regs->wdog_count);
```

Comment:
Indeed I think that using the offsets you can get rid of the `(mem_addr_t`) cast that is a bit obnoxious.

`sys_write32(0, reg_base + OFFSET)`

Reply:
Made the change.